Src/java/cql-to-elm/src/test/java/org/cqframework/cql/cql2elm/SemanticTests.java

```diff
@@ -652,6 +653,62 @@ public void testNonExistentFileName() {
         assertThrows(IOException.class, () -> TestUtils.runSemanticTest("ThisFileDoesNotExist.cql", 0));
     }
 
+    @Test
+    public void testCaseConditionalReturnTypes() throws IOException {
+        CqlTranslator translator = runSemanticTest("Issue648.cql", 0);
+        org.hl7.elm.r1.Library library = translator.toELM();
+        Map<String, ExpressionDef> defs = new HashMap<>();
+
+        if (library.getStatements() != null) {
+            for (ExpressionDef def : library.getStatements().getDef()) {
+                defs.put(def.getName(), def);
+            }
+        }
+
+        ExpressionDef caseDef = defs.get("Cases");
+    
+        assertThat(caseDef.getResultType(), instanceOf(ChoiceType.class));
+
+        ChoiceType choiceType = (ChoiceType)caseDef.getResultType();
+
+        List<String> expectedChoiceTypes = new ArrayList<>();
+        expectedChoiceTypes.add("System.String");
+        expectedChoiceTypes.add("System.Boolean");
+        expectedChoiceTypes.add("System.Integer");
+
+        for (DataType dt : choiceType.getTypes()) {
```

Comment:
Suggestion: This requirement goes both ways. I'd suggest creating two sets of strings for expected and actual and making sure they are equivalent. This test would start passing if the compiler emitted no choice types.

Reply:
The sets comparison does not respect the order of types I believe using list comparison would harden the test case. suggestion?

Reply:
The requested changes have been made.

Reply:
Set is the correct comparison here, not List.

Choice<Integer, String> ~ Choice<String,Integer>